### ISSUE TITLE
Norway and Svalbard and Jan Mayen region 

### ISSUE THREAD

**Issue by Esteban82** [GenericMappingTools/dcw-gmt#16]
For Norway, there are two ISO Codes:
NO: For the european part
[SJ](https://en.wikipedia.org/wiki/Svalbard_and_Jan_Mayen): For the Arctic islands.

The map shows the NO areas in yellow, and the SJ in red. The dataset have these errors:

- The arctic Islands are included in NO. The region of the map is based on -RNO.
- In SJ, the Jan Mayen island in NOT included (painted in yellow). 

`gmt coast -Baf -N1 -JQ15c -Sdodgerblue2 -RNO -png Norway -ENO+gyellow -ESJ+gred@50`

![Norway](https://user-images.githubusercontent.com/17908951/132574189-3bc835a4-47af-42f4-99e5-fb2d80230f9e.png)

**Comment by Esteban82** [comment]
The [Bear island](https://en.wikipedia.org/wiki/Bear_Island_(Norway)) should also be included in SJ.

**Comment by PaulWessel** [comment]
I agree, nobody in Norway expects to see Svalbard when asking for a map of Norway.

**Comment by joa-quim** [comment]
Why not? It's part of Norway. If one do Portugal we get Azores and Madeira, or Canaries for Spain.

**Comment by PaulWessel** [comment]
Yes, but it is different. Svalbard is not a part of Norway per se, we only have jurisdiction.  The Russians are there too and operating a coal mine, for instance.  Pretty sure the Italians are not operating a pizzeria in Horta.

**Comment by Esteban82** [comment]
But, for practical reasons I think it is better to split them. 
I think it would good to have a ISO code for the Canaries Island and Madeira and  Azores. For example, if someone wants to make a map of the Canaries, have to add the long and lat values. If use -RES will get a very large map.

Also, for example French Guiana is part of France. 

And if a make a map of Europe I get this


`gmt coast -Baf -N1 -JQ15c -Sdodgerblue2 -R=EU -png Europe`
![Europe](https://user-images.githubusercontent.com/17908951/132590725-8e22d30f-b3c7-4dbb-8dc4-c1bcd551c2c8.png)

While I will get this without all those islands (and maybe we could also remove Iceland). 

![Europe2](https://user-images.githubusercontent.com/17908951/132590939-ed671909-2850-40d7-866c-216e434f7a4e.png)

Or this without Iceland (which is the Atlantic Ocean).

![Europe3](https://user-images.githubusercontent.com/17908951/132591180-a81ad4ba-ef09-48a9-8ba6-b4c221e0b023.png)

**Comment by PaulWessel** [comment]
Well, Iceland really is part of Europe, it is also part of the Nordic countries, so that should not be split out.

**Comment by joa-quim** [comment]
> I think it would good to have a ISO code for the Canaries Island and Madeira and Azores.

We cannot create ISO codes. There are *official* lists of them.
Didn't look what the french dis with their *autre mer* provinces nor where is S. Helen (English).

**Comment by Esteban82** [comment]
> We cannot create ISO codes. There are official lists of them.
Yes, I agree. For the Canaries there is (IC) with the status of Exceptionally reserved.

**Comment by Esteban82** [comment]
> I agree, nobody in Norway expects to see Svalbard when asking for a map of Norway.

So, @PaulWessel how should I proceed.
1. Update the [NO.txt](https://github.com/GenericMappingTools/dcw-gmt/blob/master/orig/EU/NO.txt) and [SJ.txt ](https://github.com/GenericMappingTools/dcw-gmt/blob/master/orig/EU/SJ.txt )files. 
2. Follow the [building dcw instructions](https://github.com/GenericMappingTools/dcw-gmt#building-dcw-gmt).
3. Anything else?

**Comment by PaulWessel** [comment]
That is a good start I think

**Comment by Esteban82** [comment]
> The [Bear island](https://en.wikipedia.org/wiki/Bear_Island_(Norway)) should also be included in SJ.

I just realized that this island doesn't exist in the DCW data. What should I do?

**Comment by PaulWessel** [comment]
From experience dealing with GSHHG I think I should warn you for getting too much into details.  Once you find one missing island you will quickly find 13288 other missing islands and this is not a good use of your time to fix!
Looking at NO.txt I see that it has a duplicate of all the Svalbard polygons - looks identical to what is in SJ.txt, so I think we just need to delete that section from NO.txt.  They both seem to have 101 svalbard polygons.

This is a nicely contained bug fix more than anything, so just do that as a separate PR and this one we will merge into dcw master.

**Comment by Esteban82** [comment]
> From experience dealing with GSHHG I think I should warn you for getting too much into details. Once you find one missing island you will quickly find 13288 other missing islands and this is not a good use of your time to fix!

Ok

> Looking at NO.txt I see that it has a duplicate of all the Svalbard polygons - looks identical to what is in SJ.txt, so I think we just need to delete that section from NO.txt. They both seem to have 101 svalbard polygons.

Yes, I just did that. And move the polygon of Jan Mayen from NO.txt to SJ.txt. 

In the `config.mk` file, I just have to modified the DCW version (to 2.0.1) and the date, right? 

>     2. Follow the [building dcw instructions](https://github.com/GenericMappingTools/dcw-gmt#building-dcw-gmt).

When I run make build-dcw I get this message (those are the final lines). 

```

DCW2singleshortnc.sh: 133: let: not found
BN : Converting
DCW2singleshortnc.sh: 133: let: not found
BH : Converting
DCW2singleshortnc.sh: 133: let: not found
ID : Converting
DCW2singleshortnc.sh: 133: let: not found
Generate netCDF4...
ncgen: dcw-gmt.cdl line 3: syntax error, unexpected ';'
No such file or directory
Location: file ; line 1907

Command exited with non-zero status 1
293.07user 47.85system 5:46.67elapsed 98%CPU (0avgtext+0avgdata 204112maxresident)k
1176inputs+3375288outputs (15major+15208791minor)pagefaults 0swaps
make: *** [Makefile:106: dcw-gmt.nc] Error 1

```

Then, when I run `make archive `I get this error:

```
echo "make dcw-gmt-2.0.1.tar.gz"
make dcw-gmt-2.0.1.tar.gz
rm -f dcw-gmt-2.0.1.tar.gz
rm -rf dcw-gmt-2.0.1
mkdir -p dcw-gmt-2.0.1
cp -f dcw-gmt.nc dcw-countries.txt dcw-states.txt LICENSE README.md ChangeLog dcw-gmt-2.0.1
cp: no se puede efectuar `stat' sobre 'dcw-gmt.nc': No existe el archivo o el directorio
make: *** [Makefile:65: tar-dcw] Error 1

```

**Comment by PaulWessel** [comment]
Yep, 2.0.1 is fine for this, and the date.
I see the Makefile has a bug that does not affect me (sh is bash) but does affect you.  Please change time sh to time bash as part of your PR.  The archive error has to do with no file being created.  Making the netcdf file takes many many minutes, as you will see.

**Comment by Esteban82** [comment]
Now it works.
Later, when I run `make checksum` I got:

```
md5 -r dcw-gmt-2.0.1.tar.gz | awk '{printf "Update dcw-gmt.info with the new check sum: %s\n", $1}'
/bin/sh: 1: md5: not found
```
Is this because I don't have md5 in my pc? Is it the same that `md5sum`?

**Comment by PaulWessel** [comment]
Yes, stoopidly it is called md5 on macOS and md5sum elsewhere.  Please change to md5sum and I will make an alias for my macs.

**Comment by Esteban82** [comment]
Fixed in 2.0.1.
